### PR TITLE
[WIP] Started working on API timezone support

### PR DESCRIPTION
**Pull request by jbrooksuk**
I feel like I'm missing something pretty obvious here.

### CODE REVIEW

File: app/Http/Middleware/Timezone.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+     * @param \Closure                 $next
+     * @param string                   $type
+     *
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        if ($tz = $request->header('Time-Zone')) {
+            app('config')->set('app.timezone', $tz);
+        }
```
Comment (billmn) [inline app/Http/Middleware/Timezone.php]: This could be <code>Setting::set('app_timezone', $tz);</code> ?

**Comment by billmn** [comment]
I've tested your code locally and yes, you need to:

1. Use <code>Setting::set('app_timezone', $tz)</code>
2. Make this change to apply timezone on attributes when there is a pagination (https://github.com/cachethq/Cachet/pull/1017)

**Comment by jbrooksuk** [comment]
Point 1 is now done. Not sure how best to do point 2.

**Comment by jbrooksuk** [comment]
@billmn with the paginator stuff, do you think we should loop the results, then modify any dates:

- `deleted_at`
- `created_at`
- `updated_at`
- `scheduled_at`

And convert the timezone?

**Comment by billmn** [comment]
I have tested <code>/api/v1/incidents</code> and already works:

schermata 2015-10-08 alle 09 36 29
schermata 2015-10-08 alle 09 36 44

**Comment by jbrooksuk** [comment]
Nice! So actually, this is done? I'm always dubious about timezone code so I want to make sure this is right.

**Comment by jbrooksuk** [comment]
I need to add a test for the `Timezone` middleware really. CC @GrahamCampbell

**Pull request merged**